sympy/polys/rings.py

```diff
@@ -210,61 +209,58 @@ def __new__(cls, symbols, domain, order=lex):
         order = OrderOpt.preprocess(order)
 
         _hash_tuple = (cls.__name__, symbols, ngens, domain, order)
-        obj = _ring_cache.get(_hash_tuple)
```

Comment:
Does this still need to use `__new__` instead of just plain `__init__` with the ring cache removed?

Reply:
It could use `__init__` although we may want to use a simpler `@cache` on `__new__` at some point.

Reply:
Are the class constructors part of the API? If not I would rather do the caching in a separate function that constructs the class. It's maybe not a huge deal but avoiding `__new__` when possible leads to code that is much simpler to work with.

Reply:
It is not clear what exactly is or is not part of the API. There are many different ways of creating the `ring`.

In general I don't think that it matters whether `__new__` or `__init__` is being used unless someone is going to subclass the class. If anyone has subclassed the class though then changing from `__new__` to `__init__` is not a backwards compatible change.

Reply:
One reason to keep `__new__` is that in my prototype of adding support for python-flint's multivariate polynomials I subclass this class to have `PolynomialRing_Python` and `PolynomialRing_FLINT`. The `__new__` method here should return an instance of the appropriate subclass.